classes/ObjectModel.php

```diff
@@ -579,8 +579,10 @@ public function add($auto_date = true, $null_values = false)
             return false;
         }
 
-        // Get object id in database
-        $this->id = Db::getInstance()->Insert_ID();
+        // Get object id in database if force_id is not true
+        if (!$this->force_id || empty($this->id)) {
```

Comment:
```suggestion
        if (empty($this->id)) {
```
The `!$this->force_id` is not necessary, it's already _cleaning_ at the start of the method with:
```
if (isset($this->id) && !$this->force_id) {
    unset($this->id);
}
```

Reply:
@ytilotti indeed yes.
@matks @atomiix what do you think ?
!$this->force_id can be more clear to read and usefull if the first if is deleted, Y don't know.

Reply:
> !$this->force_id can be more clear to read and usefull if the first if is deleted, Y don't know.

If we have unit tests that run through this scenario, we dont have to be afraid "what if the first if is deleted?" because the tests would crash in that case 😉